### ISSUE TITLE
Move amp-carousel type="carousel" state to a predefined element and not only on:tap. 

### ISSUE THREAD

**Issue by andreapernici** [ampproject/amphtml#20617]
I'd like to be able to allow the automatic focus on the right part of the carousel in order to give focus to the right selected page of the horizontal scrolling menu, but looks like there's no way to combine amp.state and carousel index.

Am I missing something? Or is a non existent feature?

```
    <amp-carousel id="carousel" on="LOADACTION:carousel.goToSlide(index=9)" type="carousel">
      Menu0
      Menu1
      Menu2
      Menu3
      Menu4
      Menu5
      Menu6
      Menu7 
      Menu8
      Menu9 <!--------
....
```

Is this something I can do right now or I need to wait and improvement of the preselection?

**Comment by andreapernici** [comment]
Just to be clearer.

Imagine the following we should be able to preselect the desired menuIndex to let the horizontal menu to be positioned on the selected page.

If I'm at url /page/10 I'd like to have the "Menu10" item selected and visible in the menu.
```

<amp-state id="carousel">
  <script type="application/json">
    {
      "menuIndex": 10
    }
  </script>
</amp-state>
<amp-carousel id="carousel" [slide]="menuIndex"
        class="categories-carousel" height="40" layout="fixed-height" 
        type="carousel">
      <div class=" flex-vertical-center inline-block px3">
        <a href="#" id="oggi" [class]="menuIndex == 0 ? 'current categories-button' : 'categories-button'" class="categories-button" on="tap:AMP.setState({menuIndex: 0})">
            Menu1
        </a>
      </div>
      <div class="inline-block px3">
        <a href="#" id="domani" [class]="menuIndex == 1 ? 'current categories-button' : 'categories-button'" class="categories-button" on="tap:AMP.setState({menuIndex: 1})">
            Menu2
        </a>
      </div>
      .......

      <div class="inline-block px3">
        <a href="#" id="domani" [class]="menuIndex == 10 ? 'current categories-button' : 'categories-button'" class="categories-button" on="tap:AMP.setState({menuIndex: 10})">
            Menu2
        </a>
      </div>
    </amp-carousel>
```

**Comment by aghassemi** [comment]
Currently there is no way. `amp-bind` does not run on page load. 
We are thinking about this for carousel v2.
Related (almost dup) of https://github.com/ampproject/amphtml/issues/12659 

/to @nainar @sparhami

**Comment by andreapernici** [comment]
Yes I think it's a necessary feature considering Google is forcing everybody in using AMP, but there are still so many stuff that are far from doable with AMP.

**Comment by andreapernici** [comment]
When will it be out?

**Comment by nainar** [comment]
Hi, we are looking to launch this sometime mid Q2 2019 latest.

**Comment by sparhami** [comment]
Closing as a dupe of #20617, please re-open if there is a different feature requested.
